js/virtkeyboard.js

```diff
@@ -0,0 +1,482 @@
+  KeyG: 'g',
+  KeyH: 'h',
+  KeyJ: 'j',
+  KeyK: 'k',
+  KeyL: 'l',
+  Semicolon: ';',
+  Quote: "'",
+  KeyZ: 'z',
+  KeyX: 'x',
+  KeyC: 'c',
+  KeyV: 'v',
+  KeyB: 'b',
+  KeyN: 'n',
+  KeyM: 'm',
+  Comma: ',',
+  Period: '.',
+  Slash: '/',
+  Lang: 'En',
+};
+
+const doubleKeys = new Set();
+const keyboard = document.createElement('div');
+keyboard.className = 'keyboard';
+
+for (const item in keys) {
+  const key = document.createElement('div');
+  key.className = 'key';
+  if (
+    item === 'Backspace'
+    || item === 'Tab'
+    || item === 'ShiftLeft'
+    || item === 'ShiftRight'
+    || item === 'CapsLock'
+    || item === 'Enter'
+  ) {
+    key.classList.add('key--wide');
+  }
+  if (item === 'Space') {
+    key.classList.add('key--space');
+  }
+  key.innerHTML = keys[item];
+  key.dataset.key = item;
+  keyboard.append(key);
+}
+
+container.append(keyboard);
+
+const capsLock = document.querySelector('[data-key=CapsLock]');
+
+function changeKeys(obj) {
+  for (const item in obj) {
+    const elem = document.querySelector(`[data-key=${item}]`);
+    elem.innerHTML = obj[item];
+  }
+}
+
+function changeKeyCase(up) {
+  for (const item in keys) {
+    if (up) {
+      if (
+        item.includes('Key')
+        || item === 'Period'
+        || item === 'Comma'
+        || item === 'BracketLeft'
+        || item === 'BracketRight'
+        || item === 'Semicolon'
+        || item === 'Quote'
+      ) {
+        const elem = document.querySelector(`[data-key=${item}]`);
+        elem.innerHTML = elem.innerHTML.toUpperCase();
+      }
+    } else if (
+      item.includes('Key')
+        || item === 'Period'
+        || item === 'Comma'
+        || item === 'BracketLeft'
+        || item === 'BracketRight'
+        || item === 'Semicolon'
+        || item === 'Quote'
+    ) {
+      const elem = document.querySelector(`[data-key=${item}]`);
+      elem.innerHTML = elem.innerHTML.toLowerCase();
+    }
+  }
+}
+
+function changeShift(shiftOn) {
+  if (shiftOn) {
+    changeKeys(keysShiftOn);
+    changeKeyCase(1);
+  } else {
+    changeKeys(keysShiftOff);
+    changeKeyCase(0);
+    document.querySelector('[data-key=ShiftLeft]').classList.remove('shift');
+    document.querySelector('[data-key=ShiftRight]').classList.remove('shift');
+  }
+}
+
+function changeLang() {
+  if (document.querySelector('[data-key=Lang]').innerHTML === 'En') {
+    changeKeys(replacerRu);
+    localStorage.Lang = 'Ru';
+  } else {
+    changeKeys(replacerEn);
+    localStorage.Lang = 'En';
+  }
+}
+
+function resetSpecialKey() {
+  changeShift(0);
+  document.querySelector('[data-key=ControlLeft]').classList.remove('ctrl');
+  document.querySelector('[data-key=ControlRight]').classList.remove('ctrl');
+  document.querySelector('[data-key=AltLeft]').classList.remove('alt');
+  document.querySelector('[data-key=AltRight]').classList.remove('alt');
+}
+
+if (localStorage.Lang === 'Ru') {
+  changeLang();
+}
+
+keyboard.addEventListener('mousedown', () => {
```

Comment:
Events `mousedown` and `keydown` can use the same handler.
For example 
```javascript
if (event.target.innerHTML === 'Backspace' || event.code === 'Backspace') {
...
}
```

Reply:
!!! Code duplicate.
Because the action for the event from the keyboard and virtual keyboard should be the same it's reasonable to use the ONLY ONE handler.
Splitting it to 2 different handlers will cause problems and bugs in the future if you want to change or add other buttons.